### DISCUSSION TITLE
Keptn Helm Chart - Versionining

### DISCUSSION THREAD

**Discussion by christian-kreuzberger-dtx** [keptn/keptn#2822] [Q&A] [answered]
Right now the [Keptn Helm Chart](https://github.com/keptn/keptn/tree/master/installer/manifests/keptn) requires manual intervention for each release, e.g.:
![image](https://user-images.githubusercontent.com/56065213/102510415-e1eb7800-4087-11eb-96fa-6e7ab6f65fb5.png)
https://github.com/keptn/keptn/commit/d4f964bcb8b1835ed68a7150b8d212c4c229329f

In addition, when we package the Helm Chart, we supply the following parameters:
```
VERSION=0.7.3
helm package ${BASE_PATH}/keptn --app-version $VERSION --version $VERSION
```

I'm wondering if we can somehow automate this?
a) we could set the image versions to `latest` and overwrite them using `sed -i 's/version: latest/version: $VERSION/g' *.yaml` (edit: @warber has done something similar for the ImagePullPolicy https://github.com/warber/keptn-dev-tools/blob/03087002e405b55cb84404601eaede2e73f251eb/pull-strategy-updater/psu.sh#L31-L39 )
b) we could set values for the respective image tags in all values.yaml files
c) we could package [control-plane and delivery-plane](https://github.com/keptn/keptn/tree/master/installer/manifests/keptn/charts) in separate helm releases with their own version

Any ideas and opinions are appreciated!

**Comment by christian-kreuzberger-dtx** [comment, marked as answer]
I'll go forward and implement the first prototype using `sed` (replace the tag with the version of the actual build). This might be dangerous, but brings me to the desired result for now.

**Reply by agrimmer** [reply]
I think using `sed` is a good starting point!
However, if we would re-organize our installer into separate packages, we could use the helm CLI in order to the the `version` and `appVersion`.

**Reply by christian-kreuzberger-dtx** [reply]
For now we have implemented an automation that overwrites the repsective version / appVersion in values.yaml. Therefore we do not need to manually set those things for a release :partying_face: 

Re-organizing the installer into separate packages, especially `control-plane` vs. `non-control-plane` packages is a good idea.

IMO we should get rid of everything that's not control-plane related from the `keptn/keptn` repo (e.g., `helm`, `jmeter`). These are currently built-in services that should reside in the execution-plane. But I guess for this we will require [Uniform support](https://github.com/keptn/enhancement-proposals/pull/7) first.

@agrimmer do we have anything else to discuss? Else we can close this discussion.

**Reply by christian-kreuzberger-dtx** [reply]
As part of https://github.com/keptn/keptn/issues/3253 we will start to move continuous-delivery components away from the installer helm chart.
I'll close this discussion.
